### PR TITLE
RSS Widget

### PR DESCRIPTION
**Pull request by teizenman**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/36677

## Description
New widget for RSS feed.

## Minimum version of Cortex XSOAR
- [x] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/CommonWidgets/Scripts/RSSWidget/RSSWidget.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        return_error(str(e))
+
+    demisto.results({
+        'Type': EntryType.NOTE,
+        'ContentsFormat': EntryFormat.MARKDOWN,
+        'Contents': content,
+    })
+
+
+if __name__ in ('__builtin__', 'builtins', '__main__'):
```
Comment (yaakovi) [inline Packs/CommonWidgets/Scripts/RSSWidget/RSSWidget.py:123]: how did you create the script? it should be in the template of `demisto-sdk init`
```suggestion
if __name__ in ('__builtin__', 'builtins', '__main__'):  # pragma: no cover
```

File: Packs/CommonWidgets/Scripts/RSSWidget/RSSWidget.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    Returns:
+        str: The widget's content.
+    """
+    content: str = ''
+
+    for entry_data in entries_data:
+        content += f'## [{entry_data["title"]}]({entry_data["link"]})\n'
+        content += f'_{entry_data["timestamp"]}_\n'
+        content += f'#### {entry_data["summary"]}\n'
+        content += '---\n'
```
Comment (yaakovi) [inline Packs/CommonWidgets/Scripts/RSSWidget/RSSWidget.py:93]: ### pure suggestion
you can add the template as a constant to make it more readable:
```
TEMPLATE = '''## [{title}]({link})
_{timestamp}_
#### {summary}
---
'''
```
and here:
```suggestion
    for entry_data in entries_data:
        content += TEMPLATE.format(**entry_data)
```

what do you think? not sure which I like better

File: Packs/CommonWidgets/Scripts/RSSWidget/RSSWidget.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    try:
+        rss_raw_data = client.get_feed_data()
+        parsed_feed_data = parse_feed_data(rss_raw_data)
+        entries_data = collect_entries_data_from_response(parsed_feed_data)
+        content = create_widget_content(entries_data)
+    except Exception as e:
+        demisto.error(traceback.format_exc())
+        return_error(str(e))
+
+    demisto.results({
```
Comment (yaakovi) [inline Packs/CommonWidgets/Scripts/RSSWidget/RSSWidget.py:116]:
```suggestion
    return_results({
```

File: Packs/CommonWidgets/Scripts/RSSWidget/RSSWidget.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+            published_dt = datetime.fromtimestamp(mktime(published))
+            published_formatted = published_dt.strftime('%B %-d, %Y %-I:%M %p')
+
+            entries_data.append(
+                {
+                    'timestamp': published_formatted,
+                    'link': entry.get('link'),
+                    'title': entry.get('title'),
+                    'summary': entry.get('summary')
```
Comment (yaakovi) [inline Packs/CommonWidgets/Scripts/RSSWidget/RSSWidget.py:71]:
```suggestion
                    'summary': entry.get('summary'),
```

File: Packs/CommonWidgets/Scripts/RSSWidget/RSSWidget.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    Args:
+        feed_response (requests.Response): The raw data from the RSS feed.
+
+    Returns:
+        FeedParserDict: Parsed RSS feed data.
+    """
+    try:
+        return feedparser.parse(feed_response.text)
+    except Exception as err:
+        raise DemistoException(f"Failed to parse feed.\nError:\n{str(err)}")
```
Comment (yaakovi) [inline Packs/CommonWidgets/Scripts/RSSWidget/RSSWidget.py:41]:
```suggestion
        raise DemistoException(f"Failed to parse feed.\nError:\n{str(err)}", exception=err)
```

File: Packs/CommonWidgets/Scripts/RSSWidget/RSSWidget.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        proxy(str): Use system proxy.
+    """
+
+    def __init__(self, server_url, use_ssl, proxy):
+        super().__init__(base_url=server_url, proxy=proxy, verify=use_ssl)
+
+    def get_feed_data(self):
+        """Retrieves the data from the RSS feed.
+        """
+        return self._http_request(method='GET', resp_type='response')
```
Comment (yaakovi) [inline Packs/CommonWidgets/Scripts/RSSWidget/RSSWidget.py:26]: why are you using `resp_type='response'` and not `text`?

**Comment by yaakovi** [comment]
please attach 2 screenshots of the widget:
* working
* error

**Comment by teizenman** [comment]
Screen Shot 2021-08-22 at 14 15 11

**Comment by ShirleyDenkberg** [comment]
@yaakovi @Bargenish Doc review completed.

**Review by yaakovi — APPROVED** [review]

**Pull request merged**